arrow-data/src/data.rs

```diff
@@ -541,8 +610,9 @@ impl ArrayData {
         size
     }
 
-    /// Creates a zero-copy slice of itself. This creates a new [ArrayData]
-    /// with a different offset, len and a shifted null bitmap.
+    /// Creates a zero-copy slice of itself. This creates a new
+    /// [ArrayData] pointing at the same underlying [`Buffer`]s with a
+    /// different offset, len and a shifted null bitmap.
```

Comment:
```suggestion
    /// different offset and len
```
We don't treat the null bitmap any differently, this seems to imply we bit slice it